frontends/ol-util/src/predicates.ts

```diff
@@ -24,6 +24,26 @@ export const isNotNil = <T>(x: T): x is NonNullable<T> => {
   return true
 }
 
+type MaybeHasKeys<K extends string> = Partial<Record<K, unknown>>
+/**
+ * A curried predicate `propNames => obj => boolean`. Primarily useful because
+ * the returned function `obj => boolean` is a Typescript Predicate that can
+ * be used to filter arrays. For example:
+ *
+ * ```ts
+ * type A = { a: number }
+ * const maybes = [{ a: 1 }, {}, { a: undefined }]
+ * const definitelies: A[] = maybes.filter(propsNotNil(["a"]))
+ * ```
+ */
+export const propsNotNil = <P extends string>(propNames: P[]) => {
```

Comment:
Typescript 4.7 did not like this, but Typescript 4.8 did, which is why I upgraded.

_And how I discovered that my VSCode was using 4.8 not the version specified in our package.json...._